### PR TITLE
Add clang-tidy checking for unit tests

### PR DESCRIPTION
**Pull request by PragmaTwice**
We expand the clang-tidy checking to C++ unit tests.

### CODE REVIEW

File: tests/cppunit/cluster_test.cc [hunk 1]
```diff
@@ -113,8 +113,8 @@ TEST(Cluster, CluseterGetNodes) {
 
   std::vector<std::string> vnodes = Util::Split(output_nodes, "\n");
 
-  for (unsigned i = 0; i < vnodes.size(); i++) {
-    std::vector<std::string> node_fields = Util::Split(vnodes[i], " ");
+  for (const auto& vnode : vnodes) {
```
Comment (torwig) [inline tests/cppunit/cluster_test.cc]: Just a tiny note here: move `&` close to the `vnode` variable, not the type.

Comment (PragmaTwice) [reply tests/cppunit/cluster_test.cc]: done

**Review by torwig — DISMISSED** [review]
LGTM

**Review by git-hulk — DISMISSED** [review]

**Review by torwig — APPROVED** [review]

**Comment by PragmaTwice** [comment]
Thanks all. Merging...

**Pull request merged**